go/pkg/bertyprotocol/scenario_test.go

```diff
@@ -220,242 +383,442 @@ func testingScenario_JoinGroup(ctx context.Context, t *testing.T, pts ...*Testin
+		wg.Wait()
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+		// Check if everything is ok
+		for _, receiver := range receivers {
+			receiverID := getAccountB64PubKey(t, receiver)
+			assert.Equal(t, expectedMessagesCount, subReceivedMessagesCount[receiverID])
+		}
 
-	opts := TestingOpts{
-		Logger: testutil.Logger(t),
+		t.Logf(logTree("duration: %s", 1, false), time.Since(start))
 	}
 
-	// Setup test
-	pts, cleanup := newTestingProtocolWithMockedPeers(ctx, t, &opts, nService)
-	defer cleanup()
+	// Receivers list all expected messages
+	{
+		t.Log(logTree("Receivers List Messages (store)", 1, true))
+		start := time.Now()
 
-	pt := pts[0]
+		var wg sync.WaitGroup
+		wg.Add(len(receivers))
 
-	_, otherPK, err := crypto.GenerateEd25519Key(crand.Reader)
-	require.NoError(t, err)
+		for _, receiver := range receivers {
+			// Subscribe receivers to wait for incoming messages
+			go func(receiver *TestingProtocol) {
+				subCtx, subCancel := context.WithCancel(ctx)
+				defer subCancel()
+				defer wg.Done()
 
-	otherPKBytes, err := otherPK.Raw()
-	require.NoError(t, err)
+				req := bertytypes.GroupMessageList_Request{
+					GroupPK: groupPK,
+				}
 
-	otherSharableContact := &bertytypes.ShareableContact{
-		PK:                   otherPKBytes,
-		PublicRendezvousSeed: bytes.Repeat([]byte("."), bertytypes.RendezvousSeedLength),
-		Metadata:             []byte("useless"),
-	}
+				ml, err := receiver.Client.GroupMessageList(ctx, &req)
+				if !assert.NoError(t, err) {
+					return
+				}
 
-	_, err = pt.Client.ContactRequestSend(ctx, &bertytypes.ContactRequestSend_Request{
-		Contact: otherSharableContact,
-	})
-	require.NoError(t, err)
+				receiverID := getAccountB64PubKey(t, receiver)
 
-	gInfo, err := pt.Client.GroupInfo(ctx, &bertytypes.GroupInfo_Request{
-		ContactPK: otherPKBytes,
-	})
-	require.NoError(t, err)
-	require.NotNil(t, gInfo)
+				for {
+					if subCtx.Err() != nil {
+						return
+					}
 
-	_, err = pt.Client.ActivateGroup(ctx, &bertytypes.ActivateGroup_Request{
-		GroupPK: gInfo.Group.PublicKey,
-	})
-	require.NoError(t, err)
+					// Receive message
+					res, err := ml.Recv()
+					if err == io.EOF {
+						return
+					}
+					if !assert.NoError(t, err) {
+						continue
+					}
 
-	expectedMessages := map[string]bool{
-		"test1": false,
-		"test2": false,
-		"test3": false,
-		"test4": false,
-	}
-	expectedMessagesLock := sync.Mutex{}
-	expectedMessagesCount := len(expectedMessages)
-	expectedMessagesContent := make([]string, expectedMessagesCount)
-	i := 0
-	for msg := range expectedMessages {
-		expectedMessagesContent[i] = msg
-		i++
-	}
+					// Check if received message was expected
+					expectedMessagesLock.Lock()
+					_, expected := expectedMessages[string(res.Message)]
+					expectedMessagesLock.Unlock()
+					if !expected {
+						continue
+					}
 
-	testutil.SkipUnstable(t) // after this line, the test is unstable, especially on low hardware
```

Comment:
you should keep this

Reply:
No need to keep it anymore since the function has been split, before it was useful to skip the last part of a monolithic function that was creating accounts -> adding them as contacts -> sending/receiving messages.
Now this function only sends messages so the skip can be done upstream, on a case by case basis and in a more atomic way.

Reply:
So, now, the CI with always be green or randomly be green?

Reply:
We can have it less randomly green while having more accurate tests depending on the parts / sub-parts of the tests we chose to skip. :)